raven/src/main/java/com/getsentry/raven/buffer/DiskBuffer.java

```diff
@@ -0,0 +1,200 @@
+package com.getsentry.raven.buffer;
+
+import com.getsentry.raven.event.Event;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Arrays;
+import java.util.Iterator;
+
+/**
+ * Stores {@link Event} objects to a directory on the filesystem and allows
+ * them to be flushed to Sentry (and deleted) at a later time.
+ */
+public class DiskBuffer implements Buffer {
+
+    /**
+     * File suffix added to all serialized event files.
+     */
+    public static final String FILE_SUFFIX = ".raven-event";
+
+    private static final Logger logger = LoggerFactory.getLogger(DiskBuffer.class);
+
+    private int maxEvents;
+    private final File bufferDir;
+
+    /**
+     * Construct an DiskBuffer which stores errors in the specified directory on disk.
+     *
+     * @param bufferDir File representing directory to store buffered Events in
+     * @param maxEvents The maximum number of events to store offline
+     */
+    public DiskBuffer(File bufferDir, int maxEvents) {
+        super();
+
+        this.bufferDir = bufferDir;
+        this.maxEvents = maxEvents;
+
+        String errMsg = "Could not create or write to disk buffer dir: " + bufferDir.getAbsolutePath();
+        try {
+            bufferDir.mkdirs();
+            if (!bufferDir.isDirectory() || !bufferDir.canWrite()) {
+                logger.error(errMsg);
+            }
+        } catch (Exception e) {
+            logger.error(errMsg, e);
+        }
```

Comment:
Wouldn't we still want this block to throw (and the errors to be logged by the caller?)